tools/DevDiagnostics/DevHome.DevDiagnostics/DDApp.xaml.cs

```diff
@@ -1,113 +1,118 @@
-                services.AddSingleton<AppDetailsPage>();
-                services.AddSingleton<AppDetailsPageViewModel>();
-                services.AddSingleton<InsightsPage>();
-                services.AddSingleton<InsightsPageViewModel>();
-                services.AddSingleton<ModulesPage>();
-                services.AddSingleton<ModulesPageViewModel>();
-                services.AddSingleton<ProcessListPage>();
-                services.AddSingleton<ProcessListPageViewModel>();
-                services.AddSingleton<ResourceUsagePage>();
-                services.AddSingleton<ResourceUsagePageViewModel>();
-                services.AddSingleton<WERPage>();
-                services.AddSingleton<WERPageViewModel>();
-                services.AddSingleton<WinLogsPage>();
-                services.AddSingleton<WinLogsPageViewModel>();
-                services.AddSingleton<SettingsPage>();
-                services.AddSingleton<SettingsPageViewModel>();
-
-                // Settings sub-pages and viewmodels.
-                services.AddTransient<PreferencesViewModel>();
-                services.AddTransient<PreferencesPage>();
-                services.AddTransient<AdditionalToolsViewModel>();
-                services.AddTransient<AdditionalToolsPage>();
-                services.AddTransient<AdvancedSettingsViewModel>();
-                services.AddTransient<AdvancedSettingsPage>();
-                services.AddTransient<AboutViewModel>();
-                services.AddTransient<AboutPage>();
-            }).Build();
-
-        // Provide an explicit implementationInstance otherwise AddSingleton does not create a new instance immediately.
-        // It will lazily init when the first component requires it but the hotkey helper needs to be registered immediately.
-        Application.Current.GetService<PrimaryWindow>();
-    }
-
-    internal static ITelemetry Logger => TelemetryFactory.Get<ITelemetry>();
-
-    internal static void LogTimeTaken(string eventName, uint timeTakenMilliseconds, Guid? relatedActivityId = null) => Logger.LogTimeTaken(eventName, timeTakenMilliseconds, relatedActivityId);
-
-    internal static void LogCritical(string eventName, bool isError = false, Guid? relatedActivityId = null) => Logger.LogCritical(eventName, isError, relatedActivityId);
-
-    internal static void Log<T>(string eventName, LogLevel level, T data, Guid? relatedActivityId = null)
-        where T : EventBase
-    {
-        Logger.Log<T>(eventName, level, data, relatedActivityId ?? null);
-    }
-
-    internal static void LogError<T>(string eventName, LogLevel level, T data, Guid? relatedActivityId = null)
-        where T : EventBase
-    {
-        Logger.LogError<T>(eventName, level, data, relatedActivityId);
-    }
-
-    internal static void Log(string eventName, LogLevel level, Guid? relatedActivityId = null) => Logger.Log(eventName, level, new UsageEventData(), relatedActivityId);
-}
+// Copyright (c) Microsoft Corporation.

```

Comment:
It may be worth changing the files like this one where the whole file's line endings got corrected back to LF just for diff's sake. We can do a giant sweep to fix line endings separately.

Reply:
It's a good point. These line ends making merges/etc super painful.

For diffing, can tell the website to ignore whitespace so you can see the exact changes...

![image](https://github.com/user-attachments/assets/de2740fd-fda3-45f1-b57e-78b440d2850f)

That would give you the real diff. Now if only git merges could do that too.